wire-tests/src/commonTest/proto-kotlin/com/squareup/wire/protos/custom_options/FooBar.kt

```diff
@@ -0,0 +1,478 @@
+    label = WireField.Label.REPEATED
+  )
+  val nested: List<FooBar> = emptyList(),
+  @field:WireField(
+    tag = 101,
+    adapter = "com.squareup.wire.protos.custom_options.FooBar${'$'}FooBarBazEnum#ADAPTER"
+  )
+  val ext: FooBarBazEnum? = null,
+  @field:WireField(
+    tag = 102,
+    adapter = "com.squareup.wire.protos.custom_options.FooBar${'$'}FooBarBazEnum#ADAPTER",
+    label = WireField.Label.REPEATED
+  )
+  val rep: List<FooBarBazEnum> = emptyList(),
+  unknownFields: ByteString = ByteString.EMPTY
+) : Message<FooBar, Nothing>(ADAPTER, unknownFields) {
+  @Deprecated(
+    message = "Shouldn't be used in Kotlin",
+    level = DeprecationLevel.HIDDEN
+  )
+  override fun newBuilder(): Nothing = throw AssertionError()
+
+  override fun equals(other: Any?): Boolean {
+    if (other === this) return true
+    if (other !is FooBar) return false
+    return unknownFields == other.unknownFields
+        && foo == other.foo
+        && bar == other.bar
+        && baz == other.baz
+        && qux == other.qux
+        && fred == other.fred
+        && daisy == other.daisy
+        && nested == other.nested
+        && ext == other.ext
+        && rep == other.rep
+  }
+
+  override fun hashCode(): Int {
+    var result = super.hashCode
+    if (result == 0) {
+      result = unknownFields.hashCode()
+      result = result * 37 + foo.hashCode()
+      result = result * 37 + bar.hashCode()
+      result = result * 37 + baz.hashCode()
+      result = result * 37 + qux.hashCode()
+      result = result * 37 + fred.hashCode()
+      result = result * 37 + daisy.hashCode()
+      result = result * 37 + nested.hashCode()
+      result = result * 37 + ext.hashCode()
+      result = result * 37 + rep.hashCode()
+      super.hashCode = result
+    }
+    return result
+  }
+
+  override fun toString(): String {
+    val result = mutableListOf<String>()
+    if (foo != null) result += """foo=$foo"""
+    if (bar != null) result += """bar=$bar"""
+    if (baz != null) result += """baz=$baz"""
+    if (qux != null) result += """qux=$qux"""
+    if (fred.isNotEmpty()) result += """fred=$fred"""
+    if (daisy != null) result += """daisy=$daisy"""
+    if (nested.isNotEmpty()) result += """nested=$nested"""
+    if (ext != null) result += """ext=$ext"""
+    if (rep.isNotEmpty()) result += """rep=$rep"""
+    return result.joinToString(prefix = "FooBar{", separator = ", ", postfix = "}")
+  }
+
+  fun copy(
+    foo: Int? = this.foo,
+    bar: String? = this.bar,
+    baz: Nested? = this.baz,
+    qux: Long? = this.qux,
+    fred: List<Float> = this.fred,
+    daisy: Double? = this.daisy,
+    nested: List<FooBar> = this.nested,
+    ext: FooBarBazEnum? = this.ext,
+    rep: List<FooBarBazEnum> = this.rep,
+    unknownFields: ByteString = this.unknownFields
+  ): FooBar = FooBar(foo, bar, baz, qux, fred, daisy, nested, ext, rep, unknownFields)
+
+  companion object {
+    val FIELD_OPTIONS_FOO: FieldOptions = FieldOptions(
```

Comment:
Looks good! We should fixup indentation a bit, shouldn't be that hard - can help.